CefGlue.Common.Shared/Serialization/Deserializer.cs

```diff
@@ -0,0 +1,275 @@
+        {
+            public ReferenceInfo(object obj, JsonTypeInfo typeInfo)
+            {
+                Object = obj;
+                TypeInfo = typeInfo;
+            }
+
+            public object Object { get; private set; }
+            public JsonTypeInfo TypeInfo { get; private set; }
+        }
+
+        private static readonly IDeserializerState ListWrapperMarker = new ObjectDeserializerState(objectTypeInfo: default, objectHolder: null);
+
+        /// <summary>
+        /// Deserializes the passed jsonString argument to an instance of the TargetType.
+        /// </summary>
+        /// <typeparam name="TargetType"></typeparam>
+        /// <param name="jsonString"></param>
+        /// <param name="parametersTypes">When deserializing an argumentsArray string, it's possible to specify the specific typeInfo of each of the arguments.</param>
+        /// <returns></returns>
+        /// <exception cref="InvalidOperationException"></exception>
+        public static TargetType Deserialize<TargetType>(string jsonString, ParametersTypes parametersTypes = null)
+        {
+            try
+            {
+                if (jsonString == null)
+                {
+                    return default(TargetType);
+                }
+
+                return (TargetType)Deserialize(jsonString, typeof(TargetType), parametersTypes);
+            }
+            catch (JsonException e)
+            {
+                // wrap the json exception
+                throw new InvalidOperationException(e.Message);
+            }
+        }
+
+        private static object Deserialize(string jsonString, Type typeToConvert, ParametersTypes parametersTypes = null)
+        {
+            byte[] bytes = Encoding.UTF8.GetBytes(jsonString);
+            var reader = new Utf8JsonReader(bytes, new JsonReaderOptions() { MaxDepth = DeserializerMaxDepth });
+
+            reader.Read();
+            return Deserialize(ref reader, typeToConvert, parametersTypes);
+        }
+
+        private static object Deserialize(ref Utf8JsonReader reader, Type typeToConvert, ParametersTypes parametersTypes = null)
+        {
+            var state = new Stack<IDeserializerState>();
+            var root = new object[1];
+            var referencesMap = new Dictionary<string, ReferenceInfo>();
+            IDeserializerState previousStateIfEndArray = null;
+
+            state.Push(new ArrayDeserializerState(root, typeToConvert));
+
+            do
+            {
+                var currentState = state.Peek();
+
+                switch (reader.TokenType)
+                {
+                    case JsonTokenType.Null:
+                        currentState.SetValue(null);
+                        break;
+
+                    case JsonTokenType.Number:
+                        currentState.SetValue(
+                            reader.GetNumber(currentState.ObjectTypeInfo.GetPropertyType(currentState.PropertyName))
+                            );
```

Comment:
it makes more sense to use like this
```suggestion
                        currentState.SetValue(reader.GetNumber(currentState.GetPropertyType());
```